spec/lib/dyph3/differ/two_way_actions_spec.rb

```diff
@@ -0,0 +1,49 @@
+        end
+        # it "should show an add" do
+        #   t1  = %w[ White Grey Pink Red Brown Orange Yellow Cyan Blue Violet Invisible ]
+        #   t2 = %w[ White Grey Pink Invisible Red Brown Orange Yellow Cyan Blue Violet ]
+        #   diff = Dyph3::Differ.merge_two_way_diff(t1, t2, diff2: diff2)
+        #   binding.pry
+        #   #expect(diff.map(&:class)).to eq [Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::Add]
+        # end
+        it "should show an add" do
+          t1 = "a b c d".split
+          t2 = "a b c d e".split
+          diff = Dyph3::Differ.merge_two_way_diff(t1, t2, diff2: diff2)
+          expect(diff.map(&:class)).to eq [Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::Add]
+        end
+
+        it "should show a delete" do
+          t1 = "a b c d".split
+          t2 = "a b c".split
+          diff = Dyph3::Differ.merge_two_way_diff(t1, t2, diff2: diff2)
+          expect(diff.map(&:class)).to eq [Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::Delete]
+        end
+
+        it "should show a change" do
+          t1 = "a b c d".split
+          t2 = "a b z d".split
+          diff = Dyph3::Differ.merge_two_way_diff(t1, t2, diff2: diff2)
+          expect(diff.map(&:class)).to eq [Dyph3::Action::NoChange, Dyph3::Action::NoChange, Dyph3::Action::Delete, Dyph3::Action::Add, Dyph3::Action::NoChange]
+        end
+
+        it "should work with this real world complex example" do
+          t1 = ["#<Mido::TextBlock::Id id=\"atom_12225_brief_p_0a5374c2bb74790ed958dfb8516a76fc_0\">", "#<Mido::TextBlock::Type name=:p>", ":\"Human \"", ":\"activities \"", ":\"likely \"", ":\"caused \"", ":\"the \"", ":\"Holocene \"", ":\"mass \"", ":extinctions", ":\", \"", ":\"and \"", ":\"many \"", ":\"methods \"", ":\"have \"", ":\"been \"", ":\"employed \"", ":\"to \"", ":\"estimate \"", ":\"these \"", ":\"extinction \"", ":rates", ":\".\"", "#<Mido::TextBlock::Type closed=true name=:p>", "#<Mido::TextBlock::Id closed=true id=\"atom_12225_brief_p_0a5374c2bb74790ed958dfb8516a76fc_0\">"]
+          t2 = ["#<Mido::TextBlock::Id id=\"atom_12225_brief_p_27e015d07debd946ef199ab53ce1ca29_0\">", "#<Mido::TextBlock::Type name=:p>", ":\"Human \"", ":\"activities \"", ":\"probably \"", ":\"caused \"", ":\"the \"", ":\"Holocene \"", ":\"mass \"", ":extinctions", ":\"; \"", ":\"many \"", ":\"methods \"", ":\"have \"", ":\"been \"", ":\"employed \"", ":\"to \"", ":\"estimate \"", ":\"these \"", ":\"extinction \"", ":rates", ":\".\"", "#<Mido::TextBlock::Type closed=true name=:p>", "#<Mido::TextBlock::Id closed=true id=\"atom_12225_brief_p_27e015d07debd946ef199ab53ce1ca29_0\">"]
+          diff = Dyph3::Differ.merge_two_way_diff(t1, t2, diff2: diff2)
```

Comment:
I can rewrite the the test to check the same expected behavior, with out all the pretty printed stuff